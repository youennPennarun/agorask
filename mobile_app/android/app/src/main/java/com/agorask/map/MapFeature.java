package com.agorask.map;

import android.content.Context;

import com.facebook.react.views.view.ReactViewGroup;
import com.google.android.gms.maps.GoogleMap;

/**
 * Created by nolitsou on 10/22/16.
 */
public abstract class MapFeature extends ReactViewGroup {

    public MapFeature(Context context) {
        super(context);
    }

    public abstract Object getFeature();

    public abstract void add(GoogleMap map);

    public abstract void remove(GoogleMap map);
}
