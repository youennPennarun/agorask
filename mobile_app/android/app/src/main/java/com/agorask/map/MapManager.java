package com.agorask.map;

import android.support.annotation.Nullable;
import android.view.View;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.events.RCTEventEmitter;

import java.util.Map;

/**
 * Created by nolitsou on 10/21/16.
 */
public class MapManager extends ViewGroupManager<RNMapView> {
    public static final String REACT_CLASS = "RNMap";
    private final ReactApplicationContext reactContext;

    public MapManager(ReactApplicationContext reactContext) {
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    public RNMapView createViewInstance(ThemedReactContext context) {
        return new RNMapView(reactContext, this);
    }

    @Override
    @Nullable
    public Map getExportedCustomDirectEventTypeConstants() {
        Map<String, Map<String, String>> map = MapBuilder.of(
                "onMapReady", MapBuilder.of("registrationName", "onMapReady"),
                "onLongPress", MapBuilder.of("registrationName", "onLongPress")
        );

        return map;
    }

    @Override
    public void addView(RNMapView parent, View child, int index) {
        parent.addFeature(child, index);
    }

    @Override
    public void removeViewAt(RNMapView parent, int index) {
        parent.removeFeatureAt(index);
    }
    @Override
    public int getChildCount(RNMapView view) {
        return view.getFeatureCount();
    }
    @Override
    public View getChildAt(RNMapView view, int index) {
        return view.getFeatureAt(index);
    }

    void pushEvent(View view, String name, WritableMap data) {
        reactContext.getJSModule(RCTEventEmitter.class)
                .receiveEvent(view.getId(), name, data);
    }
}
