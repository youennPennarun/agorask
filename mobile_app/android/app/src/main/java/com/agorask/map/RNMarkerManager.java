package com.agorask.map;

import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;

import java.util.Map;

import javax.annotation.Nullable;

/**
 * Created by nolitsou on 10/22/16.
 */
public class RNMarkerManager extends ViewGroupManager<RNMarkerView> {
    public static final String REACT_CLASS = "RNMarker";

    public RNMarkerManager() {
    }

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected RNMarkerView createViewInstance(ThemedReactContext reactContext) {
        return new RNMarkerView(reactContext);
    }

    @ReactProp(name = "coordinate")
    public void setCoordinate(RNMarkerView view, ReadableMap map) {
        view.setCoordinate(map);
    }

    @Override
    @Nullable
    public Map getExportedCustomDirectEventTypeConstants() {
        Map<String, Map<String, String>> map = MapBuilder.of(
                "onPress", MapBuilder.of("registrationName", "onPress")
        );
        return map;
    }

    @Override
    public void removeViewAt(RNMarkerView parent, int index) {
        super.removeViewAt(parent, index);
        parent.update();
    }
}
