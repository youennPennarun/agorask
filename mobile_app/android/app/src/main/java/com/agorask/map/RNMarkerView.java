package com.agorask.map;

import android.content.Context;
import android.util.Log;

import com.facebook.react.bridge.ReadableMap;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;
import com.google.android.gms.maps.model.MarkerOptions;

/**
 * Created by nolitsou on 10/22/16.
 */
public class RNMarkerView extends MapFeature {

    private MarkerOptions options;
    private LatLng position;
    private Marker marker;

    public RNMarkerView(Context context) {
        super(context);
    }

    @Override
    public void add(GoogleMap map) {
        Log.d("RNMarkerView", "Add new marker");
        this.marker = map.addMarker(getOptions());
    }

    @Override
    public void remove(GoogleMap map) {
        marker.remove();
        marker = null;
    }
    public void setCoordinate(ReadableMap coordinate) {
        Log.d("RNMarkerView", "set marker coordinate");
        this.position = new LatLng(coordinate.getDouble("latitude"), coordinate.getDouble("longitude"));
        if (marker != null) {
            marker.setPosition(position);
        }
    }

    private MarkerOptions createOptions() {
        Log.d("RNMarkerView", "create marker options");
        MarkerOptions options = new MarkerOptions()
                .position(position);
        return options;
    }

    public MarkerOptions getOptions() {
        if (this.options == null) {
            this.options = createOptions();
        }
        return this.options;
    }

    public LatLng getPosition() {
        return position;
    }

    public Object getFeature() {
        return this.marker;
    }
}
