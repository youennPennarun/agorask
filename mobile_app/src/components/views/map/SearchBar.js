/* @flow */

import React, {Component, PropTypes} from 'react';
import { View, StyleSheet, Dimensions, TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import {clear, updateQuery, searchVenue} from '../../../redux/actions/search';
import {MKTextField} from 'react-native-material-kit';

import Icon from 'react-native-vector-icons/MaterialIcons';

const {width} = Dimensions.get('window');

const Textfield = MKTextField.textfield()
  .withPlaceholder('Search')
  .build();

type SearchBarPropsType = {
  query: string,
  search: Function,
  clearSearch: Function,
  updateSearchQuery: Function,
  openDrawer: Function,
};

export class SearchBar extends Component {
  static propTypes = {
    query: PropTypes.string,
    search: PropTypes.func.isRequired,
    clearSearch: PropTypes.func.isRequired,
    updateSearchQuery: PropTypes.func.isRequired,
    openDrawer: PropTypes.func.isRequired,
  };

  static defaultProps = {
    quer: '',
  };
  props: SearchBarPropsType;

  _tfRef: ?MKTextField = null;
  _onChangeText(text: string) {
    this.props.updateSearchQuery(text);
  }

  _onSubmit() {
    this.props.search();
  }
  _clear() {
    this.props.clearSearch();
  }
  _renderClearButton(): ?Object {
    if (!this.props.query) return null;
    return (
      <TouchableOpacity style={styles.rightIcon}
        onPress={() => { this._clear(); }}>
        <Icon name='close'
          size={20}
          color='#212121' />
      </TouchableOpacity>
    );
  }
  render(): Object {
    return (
      <View style={styles.container} >
        <TouchableOpacity style={styles.leftIcon}
          onPress={() => { this.props.openDrawer(); }} >
          <Icon name='menu'
            size={30}
            color='#929292' />
        </TouchableOpacity>
        <Textfield style={styles.textInput}
          ref={(tf) => { this._tfRef = tf; }}
          returnKeyType='search'
          placeholder='Search'
          blurOnSubmit
          onSubmitEditing={() => { this._onSubmit(); }}
          value={this.props.query}
          onChangeText={(text: string) => { this._onChangeText(text); }} />

        {this._renderClearButton()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: width - 20,
    height: 45,
    backgroundColor: 'white',
    marginLeft: 10,
    marginRight: 10,
    marginTop: 5,
    elevation: 2,
    flexDirection: 'row',
    alignItems: 'center',
  },
  leftIcon: {
    marginLeft: 5,
    width: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textInput: {
    height: 50,
    width: width - 100,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
  },
  rightIcon: {
    height: 50,
    width: 40,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
});

/* istanbul ignore next */
const mapStateToProps = (state: Object): Object => ({
  query: state.search.query,
});
/* istanbul ignore next */
const mapDispatchToProps = (dispatch: Function): Object => ({
  clearSearch: () => {
    dispatch(clear());
  },
  updateSearchQuery: (query) => {
    dispatch(updateQuery(query));
  },
  search: () => {
    dispatch(searchVenue());
  },
});


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchBar);
