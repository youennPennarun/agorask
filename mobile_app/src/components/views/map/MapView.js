/* @flow */
/* global navigator, alert */
import React from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';

import SearchBar from './SearchBar';

import Map from '../../natives/Map';
import MapMarker from '../../natives/MapMarker';
import {updateUserLocation} from '../../../redux/actions/user';
import {getVenuesWithTasksNearPosition} from '../../../redux/actions/venue';
import {pushRoute} from '../../../redux/actions/router';

export class MapView extends React.Component {
  state = {
    position: null,
    x: 50,
    y: 50,
  };

  componentWillMount() {
    navigator.geolocation.getCurrentPosition((position) => {
        this._onGetPosition(position);
      },
      (error) => console.log(JSON.stringify(error)),
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
    );
    this.watchID = navigator.geolocation.watchPosition(position => {
      this._onGetPosition(position);
    });
  }
  componentWillUnmount() {
    navigator.geolocation.clearWatch(this.watchID);
  }

  watchID = null;

  _onGetPosition = ({coords}) => {
    this.props.updateUserLocation({lat: coords.latitude, lng: coords.longitude});
  }
  _goToVenueDetails(venue) {
    const {goToVenueDetails} = this.props;
    goToVenueDetails(venue);
  }
  _renderMarkers(venues = []) {
    return venues.map((venue, key) => {
      const cKey = `${venue.source || 'bd'}_${venue._id || venue.foursquareId}`;
      return (
        <MapMarker key={key}
          coordinate={{
            latitude: venue.address.location[1],
            longitude: venue.address.location[0],
          }}
          onPress={() => {
            this._goToVenueDetails(venue);
          }}
          numberOfTasks={venue.nbTasks} />
      );
    });
  }
  render() {
    const {venues, searchResults, venuesError} = this.props;
    console.log('error=> ', venuesError)
    const venuesToShow = (!!searchResults) ? searchResults : venues;
    return (
      <View style={{flex: 1}}>
        <Map style={{
          ...StyleSheet.absoluteFillObject,
          backgroundColor: 'red',
        }} >
          {this._renderMarkers(venuesToShow)}
        </Map>
        <SearchBar openDrawer={this.props.openDrawer} />
      </View>
    );
  }
}

MapView.propTypes = {};

MapView.fragments = {
  venues: gql`
    fragment MapViewVenues on Venue {
      _id,
      foursquareId,
      name,
      source,
      nbTasks
      address {
        location,
      },
    }
  `,
};

const VenuesNearUserQuery = gql`
  query VenuesNearUser($lat: Float!, $lng: Float!, $radius: Float) {
    venuesWithinRadius(lat: $lat, lng: $lng, radius: $radius) {
      ...MapViewVenues
    }
  }
  ${MapView.fragments.venues}
`;

const SearchVenuesQuery = gql`
  query SearchVenues($lat: Float!, $lng: Float!, $radius: Float, $query: String!) {
    searchVenues(lat: $lat, lng: $lng, radius: $radius, query: $query) {
      ...MapViewVenues
    }
  }
  ${MapView.fragments.venues}
`;


/* istanbul ignore next */
const mapStateToProps = (state: Object): Object => ({
  userLocation: state.userLocation,
  query: state.search.query,
});
/* istanbul ignore next */
const mapDispatchToProps = (dispatch: Function): Object => ({
  goToVenueDetails: (venue) => {
    dispatch(pushRoute({
      key: 'venueDetails',
      _id: venue._id,
      id: venue._id,
      sourceId: venue.foursquareId,
      source: 'foursquare',
    }));
  },
  updateUserLocation: ({lat, lng}) => {
    dispatch(updateUserLocation(lat, lng));
  },
});


/* istanbul ignore next */
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  graphql(VenuesNearUserQuery, {
    skip: ({userLocation}): boolean => (!userLocation || !userLocation.coords),
    options: ({ userLocation: {coords} }): Object => ({
      variables: {
        lat: coords.lat,
        lng: coords.lng,
        radius: 4000,
      },
    }),
    props: ({ ownProps, data: { loading, error, venuesWithinRadius } }) => ({
      ...ownProps,
      loading,
      venuesError: error,
      venues: venuesWithinRadius,
    }),
  }),
  graphql(SearchVenuesQuery, {
    skip: ({userLocation, query}): boolean => (!query || !userLocation || !userLocation.coords),
    options: ({ userLocation: {coords}, query }): Object => ({
      variables: {
        lat: coords.lat,
        lng: coords.lng,
        radius: 4000,
        query,
      },
    }),
    props: ({ ownProps, data: { loading, error, searchVenues } }) => ({
      ...ownProps,
      loading,
      searchError: error,
      searchResults: searchVenues,
    }),
  }),
)(MapView);
