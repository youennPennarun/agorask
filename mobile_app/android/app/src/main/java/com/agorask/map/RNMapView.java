package com.agorask.map;

import android.content.Context;
import android.graphics.Point;
import android.location.Criteria;
import android.location.Location;
import android.location.LocationManager;
import android.util.Log;
import android.view.View;

import com.facebook.react.bridge.LifecycleEventListener;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.WritableNativeMap;
import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.MapView;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.model.CameraPosition;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class RNMapView extends MapView
    implements OnMapReadyCallback, GoogleMap.OnMapLongClickListener {


    private final List<MapFeature> features = new ArrayList<>();
    private final Map<Marker, RNMarkerView> markers = new HashMap<>();
    private final MapManager mapManager;
    private final ReactApplicationContext context;
    private GoogleMap map;
    public RNMapView(ReactApplicationContext reactContext,MapManager mapManager) {
        super(reactContext);
        this.context = reactContext;

        this.mapManager = mapManager;

        super.onCreate(null);
        super.onResume();
        super.getMapAsync(this);
        Log.i("RNMapView", "getMapAsync");
    }

    @Override
    public void onMapReady(final GoogleMap map) {
        Log.i("RNMapView", "onMapReady");
        this.map = map;
        map.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
            @Override
            public boolean onMarkerClick(Marker marker) {
                RNMarkerView clickedMarker = markers.get(marker);
                Log.i("RNMapView", "onMarkerClick");

                if (clickedMarker != null){
                    WritableMap event = new WritableNativeMap();
                    Log.i("RNMapView", "Push 'onPress' event");
                    mapManager.pushEvent(clickedMarker, "onPress", event);
                    return true;
                }
                return false;
            }
        });
        mapManager.pushEvent(this, "onMapReady", new WritableNativeMap());

        map.setOnMapLongClickListener(this);
        map.getUiSettings().setMyLocationButtonEnabled(true);
        map.setMyLocationEnabled(true);
        zoomOnUserLocation();

        // We need to be sure to disable location-tracking when app enters background, in-case some other module
        // has acquired a wake-lock and is controlling location-updates, otherwise, location-manager will be left
        // updating location constantly, killing the battery, even though some other location-mgmt module may
        // desire to shut-down location-services.
        LifecycleEventListener listener = new LifecycleEventListener() {
            @Override
            public void onHostResume() {
                map.setMyLocationEnabled(true);
            }
            @Override
            public void onHostPause() {
                map.setMyLocationEnabled(false);
            }

            @Override
            public void onHostDestroy() {
            }
        };

        context.addLifecycleEventListener(listener);
    }

    @Override
    public void onMapLongClick(LatLng latLng) {
        WritableMap event = new WritableNativeMap();
        event.putDouble("lat", latLng.latitude);
        event.putDouble("lng", latLng.longitude);
        Point point = map.getProjection().toScreenLocation(latLng);
        event.putDouble("x", point.x);
        event.putDouble("y", point.y);
        Log.i("RNMapView", "Push 'onLongPress' event");
        mapManager.pushEvent(this, "onLongPress", event);
    }

    public void zoomOnUserLocation() {
        if (map == null) {
            return;
        }
        LocationManager locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
        Criteria criteria = new Criteria();

        Location location = locationManager.getLastKnownLocation(locationManager.getBestProvider(criteria, false));
        if (location != null) {
            map.animateCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(location.getLatitude(), location.getLongitude()), 13));

            CameraPosition cameraPosition = new CameraPosition.Builder()
                    .target(new LatLng(location.getLatitude(), location.getLongitude()))      // Sets the center of the map to location user
                    .zoom(12)                   // Sets the zoom
                    .bearing(0)                // Sets the orientation of the camera to east
                    .build();                   // Creates a CameraPosition from the builder
            map.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition));
        }
    }

    public void addFeature(View child, int index) {
        if (child instanceof RNMarkerView) {
            RNMarkerView marker = (RNMarkerView) child;
            marker.add(map);
            features.add(index, marker);
            markers.put((Marker)marker.getFeature(), marker);
        } else {
            Log.e("RNMap", "Cannot add a child of class " + child.getClass());
        }
    }
    public void removeFeatureAt(int index) {
        Log.i("RNMapView", "removeFeatureAt " + index);
        MapFeature feature = features.remove(index);
        if (feature instanceof RNMarkerView) {
            markers.remove(feature.getFeature());
        }
        feature.remove(map);
    }
}