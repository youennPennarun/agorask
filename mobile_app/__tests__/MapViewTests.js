/* global jest, expect, describe, it, navigator */

import 'react-native';
import React from 'react';

import {MapView} from '../src/components/views/map/MapView';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

function geoMock() {
  const mock = {};
  mock.getCurrentPositionCb = () => null;
  mock.watchPositionCb = () => null;
  mock.clearWatch = jest.fn();

  mock.geolocation = {
    getCurrentPosition: (cb) => { mock.getCurrentPositionCb = cb; },
    watchPosition: (cb) => {
      mock.watchPositionCb = cb;
      return 1;
    },
    clearWatch: mock.clearWatch,
  };
  return mock;
}

it('renders correctly without any venues', () => {
  navigator.geolocation = geoMock().geolocation;
  const wrapper = shallow(
    <MapView />,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('renders correctly with venues', () => {
  navigator.geolocation = geoMock().geolocation;
  const wrapper = shallow(
    <MapView venues={[{
      _id: 'venueId1',
      name: 'venue test 1',
      address: {
        location: [50, -5],
      },
      nbTasks: 2,
    }, {
      _id: 'venueId2',
      name: 'venue test 2',
      address: {
        location: [51, -6],
      },
      nbTasks: 10,
    }, {
      _id: 'venueId3',
      name: 'venue test 3',
      address: {
        location: [52, -7],
      },
      nbTasks: 20,
    }]} />,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('renders correctly with search results', () => {
  navigator.geolocation = geoMock().geolocation;
  const wrapper = shallow(
    <MapView searchResults={[{
      foursquareId: 'foursquareVenueId1',
      source: 'foursquare',
      name: 'venue test 1',
      address: {
        location: [50, -5],
      },
      nbTasks: 2,
    }, {
      foursquareId: 'foursquareVenueId2',
      source: 'foursquare',
      name: 'venue test 2',
      address: {
        location: [51, -6],
      },
      nbTasks: 10,
    }, {
      foursquareId: 'foursquareVenueId3',
      source: 'foursquare',
      name: 'venue test 3',
      address: {
        location: [52, -7],
      },
      nbTasks: 20,
    }]} />,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('update user position called and clearWatch on componentWillUnmount', () => {
  const mock = geoMock();
  navigator.geolocation = mock.geolocation;
  const updateUserLocation = jest.fn();
  const wrapper = shallow(
    <MapView updateUserLocation={updateUserLocation} />,
    { lifecycleExperimental: true },
  );
  mock.getCurrentPositionCb({ coords: {latitude: 50, longitude: -5}});
  expect(updateUserLocation.mock.calls.length).toBe(1);
  mock.watchPositionCb({ coords: {latitude: 50, longitude: -5}});
  expect(updateUserLocation.mock.calls.length).toBe(2);
  wrapper.unmount();
  expect(mock.clearWatch.mock.calls.length).toBe(1);
});
