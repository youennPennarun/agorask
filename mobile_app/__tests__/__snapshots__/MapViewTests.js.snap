exports[`test renders correctly with search results 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <MapWrapper
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }>
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -5,
          "longitude": 50,
        }
      }
      numberOfTasks={2}
      onPress={[Function]} />
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -6,
          "longitude": 51,
        }
      }
      numberOfTasks={10}
      onPress={[Function]} />
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -7,
          "longitude": 52,
        }
      }
      numberOfTasks={20}
      onPress={[Function]} />
  </MapWrapper>
  <Connect(SearchBar) />
</View>
`;

exports[`test renders correctly with venues 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <MapWrapper
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }>
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -5,
          "longitude": 50,
        }
      }
      numberOfTasks={2}
      onPress={[Function]} />
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -6,
          "longitude": 51,
        }
      }
      numberOfTasks={10}
      onPress={[Function]} />
    <MarkerWrapper
      coordinate={
        Object {
          "latitude": -7,
          "longitude": 52,
        }
      }
      numberOfTasks={20}
      onPress={[Function]} />
  </MapWrapper>
  <Connect(SearchBar) />
</View>
`;

exports[`test renders correctly without any venues 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <MapWrapper
    style={
      Object {
        "backgroundColor": "red",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    } />
  <Connect(SearchBar) />
</View>
`;
